lib/librato/metrics/queue.rb

```diff
@@ -118,31 +136,36 @@ def queued
         return {} if @queued.empty?
         globals = {}
         globals[:source] = @source if @source
+        globals[:tags] = @tags if has_tags?
         globals[:measure_time] = @measure_time if @measure_time
+        globals[:time] = @time if @time
         @queued.merge(globals)
       end
 
       # Count of metrics currently queued
       #
       # @return Integer
       def size
-        self.queued.inject(0) { |result, data| result + data.last.size }
+        self.queued.reject { |key| key == :multidimensional }.inject(0) { |result, data| result + data.last.size }
```

Comment:
Agreed. Removed the `multidimensional` flag. I actually think the persister could be simplified even further by using `/v1/metrics` if gauges or counters are present, otherwise use `/v1/measurements`.